experiment/simulation/js/main.js

```diff
@@ -1 +1,272 @@
+
+        brokenBoxPart2 = [
+            [plateStartX + plateWidth - boxWidth + breakLength + gap, boxStarty],
+            [plateStartX + boxWidth + gap / 2, boxStarty],
+            [plateStartX + boxWidth + gap / 2, bottomPlateStartY],
+            [plateStartX + boxWidth - breakLength + gap, bottomPlateStartY],
+        ];
+
+        bench = [
+            [0, canvas.height - benchHeight],
+            [canvas.width, canvas.height - benchHeight],
+            [canvas.width, canvas.height],
+            [0, canvas.height],
+        ]
+
+        step = 0;
+
+        document.getElementById("stress").innerHTML = "0.0 Mpa";
+        document.getElementById("strain").innerHTML = "0.0";
+        document.getElementById("load").innerHTML = "0.0 N";
+        document.getElementById("deformation").innerHTML = "0.0 mm";
+    }
+
+    function restart() {
+        window.clearTimeout(tmHandle);
+        setAll();
+        graph();
+        play();
+    }
+
+    function play() {
+        tmHandle = window.setTimeout(draw, 1000 / fps);
+        pauseButton.removeAttribute("disabled");
+        restartButton.removeAttribute("disabled");
+        playButton.setAttribute("disabled", "true");
+    }
+
+    function pause() {
+        window.clearTimeout(tmHandle);
+        pauseButton.setAttribute("disabled", "true");
+        playButton.removeAttribute("disabled");
+    }
+
+    function drawObject(ctx, obj, color) {
+        ctx.save();
+        ctx.fillStyle = color;
+        ctx.strokeStyle = color;
+        ctx.beginPath();
+        ctx.moveTo(obj[0][0], obj[0][1]);
+
+        for (let i = 0; i < obj.length; ++i) {
+            const next = (i + 1) % obj.length;
+            ctx.lineTo(obj[next][0], obj[next][1]);
+        }
+
+        ctx.closePath();
+        ctx.fill();
+        ctx.stroke();
+        ctx.restore();
+    }
+
+    function move(tube, flag) {
+        for (let i = 0; i < tube.length; ++i) {
+            tube[i][1] += flag;
+        }
+    }
+
+    let upperPlate = [];
+    let bottomPlate = [];
+    let upperRod = [];
+    let lowerRod = [];
+    let box = [];
+    let brokenBoxPart1 = [];
+    let brokenBoxPart2 = [];
+    let bench = [];
+    let tmHandle;
+    let chart;
+    let step = 0;
+
+
+
+    const canvas = document.getElementById("main");
+    canvas.width = 900;
+    canvas.height = 1200;
+    canvas.style = "border:3px solid;";
+    const ctx = canvas.getContext("2d");
+
+    const fill = "black";
+    const lineWidth = 1.5;
+    const originalfps = 20;
+    let fps = 20;
+
+
+    const plateStartX = 250;
+    const plateWidth = 400;
+    const upperPlateStartY = 340;
+    const bottomPlateStartY = 700;
+    const plateHeight = 120;
+    const rodWidth = 240;
+    const upperRodHeight = 300;
+    const lowerRodHeight = 360;
+    const boxWidth = 320;
+    const boxStarty = 500;
+    const breakLength = 80;
+    const gap = 10;
+    const benchHeight = 100;
+
+    const boxColor = "#2C9AD1"; // virtual labs color
```

Comment:
Please take out the color codes and put them in a separate file with mnemonic keys. Look here for example:
https://github.com/virtual-labs/exp-compaction-test-iiith/blob/main/experiment/simulation/js/data.json